osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs

```diff
@@ -76,15 +76,18 @@ public async Task<List<CarouselItem>> Run(IEnumerable<CarouselItem> items, Cance
                     {
                         var beatmap = (BeatmapInfo)item.Model;
 
-                        if (BeatmapSetsGroupedTogether)
+                        bool newBeatmapSet = lastBeatmap?.BeatmapSet!.ID != beatmap.BeatmapSet!.ID;
+
+                        if (newBeatmapSet)
                         {
-                            bool newBeatmapSet = lastBeatmap?.BeatmapSet!.ID != beatmap.BeatmapSet!.ID;
+                            if (!setMap.TryGetValue(beatmap.BeatmapSet!, out currentSetItems))
+                                setMap[beatmap.BeatmapSet!] = currentSetItems = new HashSet<CarouselItem>();
+                        }
```

Comment:
This means that when beatmap sets are not grouped together, `setMap` / `SetItems` will always link to the first panel encountered, meaning that `selectRecommendedDifficultyForBeatmapSet()` will always select that one panel. I do not think that's how it should be, and is probably the reason for the failing test here.

Reply:
I have a hard struggle understanding what you are saying here, so just attempted to fix the issues based on test failures and what seemed wrong. Can you double check and see if your concerns are also alleviated by the change?

Reply:
i dunno at this point either, probably